client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -958,15 +966,59 @@ public final void fieldCapsAsync(FieldCapabilitiesRequest fieldCapabilitiesReque
                 response -> parseEntity(response.getEntity(), entityParser), ignores);
     }
 
+    /**
+     * Defines a helper method for performing a request and then parsing the returned entity using the provided entityParser.
+     */
+    protected final <Req extends Validatable, Resp> Resp performRequestAndParseEntity(Req request,
+                                                                  CheckedFunction<Req, Request, IOException> requestConverter,
+                                                                  RequestOptions options,
+                                                                  CheckedFunction<XContentParser, Resp, IOException> entityParser,
+                                                                  Set<Integer> ignores) throws IOException {
+        return performRequest(request, requestConverter, options,
+                response -> parseEntity(response.getEntity(), entityParser), ignores);
+    }
+
+    /**
+     * @deprecated If creating a new HLRC ReST API call, consider creating new actions instead of reusing server actions. The Validation
+     * layer has been added to the ReST client, and requests should extend
+     * {@link org.elasticsearch.protocol.xpack.common.ValidationException} instead of `ActionRequest`.
```

Comment:
Shouldn't this link to  `... Validatable` not `... ValidationException`. Same with the other comments

Reply:
DOH!

Reply:
fixed in d0ca91cd2871464cc160672a79f24b2006080d40